components/common-go/baseserver/server_test.go

```diff
@@ -114,7 +114,7 @@ func TestServer_Metrics_gRPC(t *testing.T) {
 
 func TestServer_Metrics_HTTP(t *testing.T) {
 	srv := baseserver.NewForTests(t, baseserver.WithHTTP(&baseserver.ServerConfiguration{
-		Address: fmt.Sprintf("localhost:%d", baseserver.MustFindFreePort(t)),
+		Address: "localhost:0",
```

Comment:
we can use `MustUseRandomLocalAddress` here, no?

Reply:
I took this test as testing the underlying configuration options (ie, how baseserver works with `baseserver.ServerConfiguration`) and for that reason didn't want to use the helpers which themselves generate the config. But it shouldn't matter.